dominion/tasks.py

```diff
@@ -12,45 +12,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License
 
+import time
+
 from celery.utils.log import get_task_logger
+from docker.errors import APIError, NotFound
 from dominion.app import APP
-from dominion.base import DOCKER_CLIENT, BaseBuildTask
-from dominion.settings import QUEUE_BUILD_NAME
+from dominion.base import DOCKER, BaseBuildTask
+from dominion.settings import (
+    CONTAINER_NAME,
+    POLLING_FREQUENCY,
+    QUEUE_BUILD_NAME,
+    QUEUE_WATCH_NAME,
+    TIMEOUT,
+)
 from images.models import Image
 
-from dominion.util import connect_to_redis, terminalize
+from dominion.util import check_exit_code, connect_to_redis, terminalize
 
 ENV = {
     'CR': 'false',
     'N': 30,
     'TERM': 'xterm',
 }
 
+EXIT_STATUS = 'exited'
+
 LOGGER = get_task_logger(__name__)
 
 
 @APP.task(bind=True, base=BaseBuildTask)
 def build(self, image_id):
     """Builds an image. """
 
+    container_name = CONTAINER_NAME.format(image_id=image_id)
+    LOGGER.info(f'Running {container_name}')
+
     self.request.kwargs['image_id'] = image_id
 
     env = ENV.copy()
 
-    LOGGER.info(f'Start building {image_id}')
-
-    channel_name = f'build-log-{image_id}'
-    container_name = f'pieman-{image_id}'
-    conn = connect_to_redis()
     run_kwargs = {
         'detach': True,
         'name': container_name,
         'environment': env,
     }
-    self.request.kwargs['container'] = DOCKER_CLIENT.containers.run('count-von-count', **run_kwargs)
-    for line in self.request.kwargs['container'].logs(follow=True, stream=True):
+    container = self.request.kwargs['container'] = DOCKER.run('count-von-count', **run_kwargs)
+
+    watch.apply_async((image_id, ), queue=QUEUE_WATCH_NAME)
+
+    channel_name = f'build-log-{image_id}'
```

Comment:
`channel_name` pattern might be worth moving to the settings, as `CONTAINER_NAME`.

Reply:
Agreed. But we don't touch it here, so let's move it to the settings when it's really necessary.